content/ADR-145-pick-realm-v2.md

```diff
@@ -59,46 +79,137 @@ https://github.com/decentraland/kernel/blob/main/packages/shared/dao/index.ts#L6
-As Content Server and Lambdas share the same version for each Catalyst and they are using semver, there is always an unique way to sort the candidates.
-
+This rule will have multiple Catalysts, sorted by priority, defined in its configuration. The rule will attempt to connect to the first Catalyst in the list. If it is unavailable, it will continue trying with the subsequent Catalysts specified in the configuration in order of priority.
 
 _Pseudo Code_:
 
 ```typescript
-export function versionLink() {
-  const catalystsWithVersion: { realm: string, version: string } = peers.sortByVersion()
-  const versionBuckets: PriorityQueue<{ realm: string, version: string }[]> = groupByVersions(catalystsWithVersion)
-  // This selects randomly between the candidates in the selected bucket 
-  // And chooses the bucket with the same algorithm used for close peers
-  return selectFirstByScore(context, score, definitiveDecisionThreshold)
+export function catalystVersionLink(configuration: ForceCatalystConfig) {
+    let picked = context.picked
+    
+    const candidates = picked.map(candidate => candidate.catalystName)
+    const selected = configuration.sortedOptions?.find(candidate => candidates.includes(candidate))
+    
+    context.selected = !!selected 
+        ? picked.find(candidate => candidate.catalystName === selected)
+        : undefined
+
+    return context
 }
 ```
+
+_Configuration_:
+
+```typescript
+export type ForceCatalystConfig = {
+  config?: { sortedOptions?: [string]; }
+}
+```
+
+### Large Latency
+
+_Description_:
+
+This rule will still operate as it currently does. However, in order to implement it on the server-side, the client will need to check the latency on all nodes to pass this information to the service. To prevent clients from constantly calling all nodes to check latency, they will first check if the realm-picking service is using a `variant` that contains the `LARGE_LATENCY` filter and only if it does it will gather the latency information to pass it over.
+
+### Variants
+
+#### Default
+
+This variant will serve as the default configuration when no specific context needs to be addressed.
+- Filter by OVERLOADED_CATALYST
+  - Firstly, nodes that are struggling to perform as expected will be filtered out.
+- Filter by LARGE_LATENCY
+  - Next, nodes that are located far away from the user in terms of latency will be filtered out.
+- Score by CLOSE_PEERS_SCORE
+  - After filtering, if the user is attempting to connect to a specific scene, they will be redirected to the node with the highest number of peers in that scene.
+- Score by ALL_PEERS_SCORE
+  - If the previous score does not result in the selection of a single catalyst, the node with the highest number of peers connected to it will be chosen.
+- Balance with LOAD_BALANCING
+  - As a fallback, if multiple catalysts are eligible for selection, a Round-Robin strategy will be implemented to evenly distribute the load among them.
```

Comment:
nit:

```suggestion
1. Filter by OVERLOADED_CATALYST
  - Firstly, nodes that are struggling to perform as expected will be filtered out.
2. Filter by LARGE_LATENCY
  - Next, nodes that are located far away from the user in terms of latency will be filtered out.
3. Score by CLOSE_PEERS_SCORE
  - After filtering, if the user is attempting to connect to a specific scene, they will be redirected to the node with the highest number of peers in that scene.
4. Score by ALL_PEERS_SCORE
  - If the previous score does not result in the selection of a single catalyst, the node with the highest number of peers connected to it will be chosen.
5. Balance with LOAD_BALANCING
  - As a fallback, if multiple catalysts are eligible for selection, a Round-Robin strategy will be implemented to evenly distribute the load among them.
```